mithril-aggregator/src/http_server/routes/artifact_routes/cardano_database.rs

```diff
@@ -0,0 +1,367 @@
+use crate::http_server::routes::middlewares;
+use crate::http_server::routes::router::RouterState;
+use warp::Filter;
+
+pub fn routes(
+    router_state: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    artifact_cardano_database_list(router_state)
+        .or(artifact_cardano_database_by_id(router_state))
+        .or(serve_cardano_database_dir(router_state))
+}
+
+/// GET /artifact/cardano-database
+fn artifact_cardano_database_list(
+    router_state: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::path!("artifact" / "cardano-database")
+        .and(warp::get())
+        .and(middlewares::with_logger(router_state))
+        .and(middlewares::with_http_message_service(router_state))
+        .and_then(handlers::list_artifacts)
+}
+
+/// GET /artifact/cardano-database/:id
+fn artifact_cardano_database_by_id(
+    dependency_manager: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::path!("artifact" / "cardano-database" / String)
+        .and(warp::get())
+        .and(middlewares::with_logger(dependency_manager))
+        .and(middlewares::with_http_message_service(dependency_manager))
+        .and(middlewares::with_metrics_service(dependency_manager))
+        .and_then(handlers::get_artifact_by_signed_entity_id)
+}
+
+fn serve_cardano_database_dir(
+    router_state: &RouterState,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+    warp::path("cardano_database_download")
```

Comment:
I'm not sure to understand why the route isn’t specified in the OpenAPI file. Could you explain it to me, please? 🙏

Reply:
This route is here mainly for testing the artifacts in the e2e test.
IMO we should create a separate server for that purpose which would avoid having no to publish the route in the OpenAPI.